ibis/file/client.py

```diff
@@ -0,0 +1,83 @@
+import ibis
+import ibis.expr.types as ir
+from ibis.pandas.execution import execute
+from ibis.file.utils import pathlib
+import ibis.file.execution  # noqa
+
+
+class Client(ibis.client.Client):
+
+    def __init__(self, root):
+        super(Client, self).__init__()
+        self.root = pathlib.Path(str(root))
+        self.dictionary = {}
+
+    def database(self, name=None, path=None):
+        if name is None:
+            return Database('root', self, path=path)
+
+        if name not in self.list_databases(path):
+            raise AttributeError(name)
+        if path is None:
+            path = self.root
+
+        new_name = "{}.{}".format(name, self.extension)
+        if (self.root / name).is_dir():
+            path = path / name
+        elif not str(path).endswith(new_name):
+            path = path / new_name
+
+        return Database(name, self, path=path)
+
+    def table(self, name, path):
+        raise NotImplementedError
+
+    def execute(self, expr, params=None, **kwargs):
+        assert isinstance(expr, ir.Expr)
+        return execute(expr, params=params)
+
+    def list_tables(self, path=None):
+        raise NotImplementedError
+
+    def list_databases(self, path=None):
+        raise NotImplementedError
+
+
+class Database(ibis.client.Database):
+
+    def __init__(self, name, client, path=None):
+        super(Database, self).__init__(name, client)
+        self.path = path
+
+    def __str__(self):
+        return '{0.__class__.__name__}({0.name})'.format(self)
+
+    def __dir__(self):
+        dbs = self.list_databases(path=self.path)
+        tables = self.list_tables(path=self.path)
+        return sorted(list(set(dbs).union(set(tables))))
```

Comment:
You don't need to sort this.

Reply:
you don't but its nicer to be consistent on output

Reply:
Right, I mean that it's already sorted for you so the consistency is there without having to provide it ourselves:

IPython:

```python
In [3]: class Foo:
   ...:     def __dir__(self):
   ...:         return ['b', 'a']
   ...: 

In [4]: foo = Foo()

In [5]: dir(foo)
Out[5]: ['a', 'b']
```

Vanilla Python:

```python
>>> class Foo:
...     def __dir__(self):
...         return ['b', 'a']
... 
>>> foo = Foo()
>>> dir(foo)
['a', 'b']
```

Reply:
Ok, I'm not sure why this appears to always sort, but it appears to work on my machine and not necessarily others.

@jreback Keeping `sorted` is fine, but can you remove the call to `list` since `sorted` always returns a `list`?